src/ComponentKernelImplementation.java

```diff
@@ -0,0 +1,180 @@
+package components.setlist;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Kernel implementation of ConcertSetlist.
+ *
+ * Convention: The internal list `setlist` contains only unique song titles.
```

Comment:
Make sure you use the @ tags
```suggestion
 * @convention the internal list `setlist` contains only unique song titles.
```